libs/sdk-ui/src/base/react/usePromiseCache.ts

```diff
@@ -43,8 +43,11 @@ export function usePromiseCache<TParams, TResult, TError>(
 
     useEffect(() => {
         return () => {
-            promiseCacheRef.current.reset();
-            setInitialState();
+            if (!shouldResetPromises) {
```

Comment:
This condition looks reversed
Reading the code like "_if not should reset promises -> do the reset_" could be confusing. Please check and consider better naming.

Reply:
I have updated this. Thank you!